scripts/tasks.sh

```diff
@@ -900,6 +908,45 @@ function flannel_images_present() {
     fi
 }
 
+# Determine if the cluster is eligible for a multi-node migration and then initiate the migration
+function migrate_to_multinode_storage() {
+    export KUBECONFIG=/etc/kubernetes/admin.conf
+
+    # Is Rook and OpenEBS installed
+    if ! kubectl get ns | grep -q rook-ceph && ! kubectl get ns | grep -q openebs; then
+        bail "Rook and OpenEBS must be installed in order to migrate to multi-node"
+    fi
+
+    # Get Rook.minimumNodeCount option from the configmap
+    local rookMinNodes=
+    rookMinNodes=$(kubectl get cm kurl-current-config -n kurl -ojsonpath='{.data.addons-rook}' | base64 -d | tr "," " "| awk '{print $1}' | cut -d ":" -f 2)
+    if [ -z "$rookMinNodes" ] || [ "$rookMinNodes" -le "1" ]; then
+        bail "Rook.MinimumNodeCount must be greater than 1"
+    fi
+
+    local ekcoVersion=
+    local requiredEkcoVersion=
+    requiredEkcoVersion="0.27.1"
+    ekcoVersion=$(kubectl get cm kurl-current-config -n kurl -ojsonpath='{.data.addons-ekco}' | base64 -d | tr "," " "| awk '{print $1}' | cut -d ":" -f 2 | tr -d '}"')
+    semverCompare "$ekcoVersion" "$requiredEkcoVersion"
+    if [ "$SEMVER_COMPARE_RESULT" -lt "0" ]; then
+        bail "EKCO add-on version >= $requiredEkcoVersion required for migration"
+    fi
+
+    local ekcoAddress=
+    local ekcoAuthToken=
+    ekcoAddress=$(get_ekco_addr)
+    ekcoAuthToken=$(get_ekco_storage_migration_auth_token)
+
+    # Initiate OpenEBS to Rook multi-node migration
+    if ! "${DIR}"/bin/kurl cluster migrate-multinode-storage --ekco-address "$ekcoAddress" --ekco-auth-token "$ekcoAuthToken" < /dev/tty; then
```

Comment:
you may want to wrap this in the `prompts_can_prompt` function. also if the prompt should default to no we have the flag `yes` that you should support.

Reply:
my question still stands here. what happens if there is no terminal? say this is invoked from automation.

Reply:
this should really share code with rook_maybe_migrate_from_openebs too

Reply:
If there is no TTY I would think we should default to _not_ initiate the migration and issue a message with the command which the user can then use to trigger the migration at a later time. For automation purposes, we could also add a `-y`/`--asume-yes` flag to the `cluster migrate-multinode-storage` command.